### DISCUSSION TITLE
建议停止支持winxp等老旧系统

### DISCUSSION THREAD

**Discussion by chen-qingyu** [xmake-io/xmake#4425] [General]
此处的“老旧系统”指的是 WinXP等官方早已停止维护的操作系统 或 32位的个人电脑操作系统，嵌入式单片机不在此列。
原因：
1. 作者时间精力有限，将宝贵的时间精力放在现代操作系统上更值得
2. 使用winxp等系统的人大概也不使用现代构建工具
3. 精简整个构建脚本，提高速度

**Comment by waruqi** [comment]
？？？我好像也没写多少额外的代码 去特地支持 xp、32，就算删掉所有跟 xp 支持相关的脚本代码，也不超过 10 行。。何来耗费精力一说呢？精简掉那10几行脚本，也不会变快多少。

我都快一年没特别维护过 xp 32 相关的东西了，还不是跑的好好的，既然它跑的很好，我也没花精力，干嘛要停掉支持呢。。而且我自己 xp 还在用呢，个人mac本子 磁盘空间有限，虚拟机跑 xp ，日常做做 win 下开发调试，效率比 win11 高多了，还轻量，不卡，我一直在用。。

> 使用winxp等系统的人大概也不使用现代构建工具

我不就一直在用么。。 = =

**Comment by waruqi** [comment]
不像 ninja 那种，所有系统 api 的封装和不同系统版本 接口的兼容性 处理都是 ninja 内部处理，各种判断，硬编码处理，维护兼容 xp 等老系统会很麻烦，所以抛弃了支持，它里面到处都是各种 #if/else 判断

xmake 依托于 跨平台的 tbox 底层平台库，已经抽象了一层，对于 xmake 来说，真的没有多少需要特别对 xp 处理的地方了

也就处理编译 xmake binary 加了一个兼容 xp 的 flags，让它能在 xp 上可运行而已。

https://github.com/xmake-io/xmake/blob/8865866a47f023f4b36453d25c75e876f503323d/core/src/demo/xmake.lua#L49

好像也没啥了。。而 tbox 内部，win 下所有 api 的封装，全是走动态库加载的，自适应切换，也不会特别去判断是否 xp 平台支持，如果当前系统 这个 win api 不可用，会自动切换到其他的实现方案，这块封装完，基本几年都不需要改动，你可以看 tbox 的commit ，win 相关的兼容性方面的改动，几年也没改过几次

**Reply by chen-qingyu** [reply]
噢噢……这样啊，那看来是我想多了
